doc/design/switch_kernel.md

```diff
@@ -0,0 +1,66 @@
+## Background
+Every operator has many kernels. It because there are multiple data types, places, data layout that Fluid supports. We use the `KernelType` to describe kernel types that operators can hold. 
+
+The `KernelType` is as follows.
+
+```
+struct KernelType {
+  Place place_;
+  DataType data_type_;
```

Comment:
Some operators have 2 types, like cost and evaluators, they must have label input which is sometimes int64.

Reply:
The below `major data_type` has considered this situation.

Reply:
Yes. We have some operators which take two or more than two kinds of data types as their inputs.

However, I think it is easy to use a single data type for each operator. Even for the `cost` and `evaluators`, each of them has a single data type for output. And the output data type of them is decided by one input.

For example, `cross_entropy` has `logit` and `label` as its inputs, and produces `loss` as its output. The data type of `loss` is same as `logit`. So we think the data type of cross_entroy is `float` when `logit` is `float`.

The `label` can not only be `int64`, but also can be `int32`. We can add a manually switch in kernel object to achieve this.

Reply:
I see, thank you guys for detailed explains.